apisix/stream/xrpc/runner.lua

```diff
@@ -27,6 +27,19 @@ local pcall = pcall
 local ipairs = ipairs
 local tostring = tostring
 
+
+core.ctx.register_var("rpc_time", function(ctx)
+    local session = ctx.xrpc_session
+    local curr_ctx_id = session._currr_ctx_id
+    local curr_ctx = session._ctxs[curr_ctx_id]
```

Comment:
I think we can implement a special var meta for the xRPC ctx?
A new version of https://github.com/apache/apisix/blob/4690feb421779f5b79e8dd990dc00f4d3f1052d0/apisix/core/ctx.lua#L216.

Therefore we can avoid looking up current xRPC ctx in every vars.

BTW, the `release_vars` is not called for xRPC ctx.

Reply:
Or we can pass ctx in https://github.com/apache/apisix/blob/4690feb421779f5b79e8dd990dc00f4d3f1052d0/apisix/core/ctx.lua#L343 as var._ctx

Reply:
due to:https://github.com/apache/apisix/blob/4690feb421779f5b79e8dd990dc00f4d3f1052d0/apisix/core/ctx.lua#L345

in `core.ctx.register_var`, the `ctx.var._ctx` would point to `ngx.ctx.api_ctx`, so I choose  to pass th xRPC ctx in `session._curr_ctx`

Reply:
@tzssangglass 
The `register_var` doesn't rewrite the `ctx.var`. `ctx.var._ctx` will point to the ctx passed via `set_vars_meta(ctx)`.